File: example1/src/main/java/com/example/grpc/echo/unary/UnaryServer.java
package com.example.grpc.echo.unary;

import com.example.grpc.echo.EchoRequest;
import com.example.grpc.echo.EchoResponse;
import com.example.grpc.echo.EchoServiceGrpc;
import com.example.grpc.Servers;
import io.grpc.stub.StreamObserver;

import java.util.logging.Logger;

public class UnaryServer {

    private static final Logger logger = Logger.getLogger(UnaryServer.class.getName());

    public static void main(String[] args) throws Exception {
        Servers.start(new EchoServiceImpl(), logger);
    }

    private static class EchoServiceImpl extends EchoServiceGrpc.EchoServiceImplBase {
        @Override
        public void unaryEcho(EchoRequest request, StreamObserver<EchoResponse> responseObserver) {
            logger.info("request: "+ request.getMessage());
            var response = EchoResponse.newBuilder().setMessage("hello " + request.getMessage()).build();
            responseObserver.onNext(response);
            responseObserver.onCompleted();
        }
    }
}

File: _kotlin/src/main/java/com/example/grpc/echo/unary/UnaryServer.kt
package com.example.grpc.echo.unary

import com.example.grpc.EchoRequest

object  /*TODO*/ UnaryServer {
    private val logger: java.util.logging.Logger = java.util.logging.Logger.getLogger(UnaryServer::class.java.getName())

    @kotlin.Throws(java.lang.Exception::class)
    @kotlin.jvm.JvmStatic
    fun main(args: kotlin.Array<kotlin.String>) {
        Servers.start(EchoServiceImpl(), UnaryServer.logger)
    }

    private class  /*TODO*/ EchoServiceImpl : EchoServiceGrpc.EchoServiceImplBase() {
        public override fun unaryEcho(request: EchoRequest, responseObserver: StreamObserver<EchoResponse?>) {
            UnaryServer.logger.log(java.util.logging.Level.INFO, "request: {0}", request.getMessage())
            val response: Unit /* TODO: class org.jetbrains.kotlin.nj2k.types.JKJavaNullPrimitiveType */? =
                EchoResponse.newBuilder().setMessage("hello " + request.getMessage()).build()
            responseObserver.onNext(response)
            responseObserver.onCompleted()
        }
    }
}
